labelbox/schema/dataset.py

```diff
@@ -749,3 +773,106 @@ def _export(
+        >>>         {
+        >>>             "row_data": "http://my_site.com/photos/img_01.jpg",
+        >>>             "global_key": "global_key1",
+        >>>             "external_id": "ex_id1",
+        >>>             "attachments": [
+        >>>                 {"type": AttachmentType.RAW_TEXT, "name": "att1", "value": "test1"}
+        >>>             ],
+        >>>             "metadata": [
+        >>>                 {"name": "tag", "value": "tag value"},
+        >>>             ]
+        >>>         },
+        >>>         # update existing data row by global key
+        >>>         {
+        >>>             "global_key": "global_key1",
+        >>>             "external_id": "ex_id1_updated"
+        >>>         },
+        >>>         # update global key of data row by existing global key
+        >>>         {
+        >>>             "key": GlobalKey("global_key1"),
+        >>>             "global_key": "global_key1_updated"
+        >>>         },
+        >>>         # update data row by ID
+        >>>         {
+        >>>             "key": UniqueId(dr.uid),
+        >>>             "external_id": "ex_id1_updated"
+        >>>         },
+        >>>     ])
+        >>>     task.wait_till_done()
+        """
+        if len(items) > MAX_DATAROW_PER_API_OPERATION:
+            raise MalformedQueryException(
+                f"Cannot upsert more than {MAX_DATAROW_PER_API_OPERATION} DataRows per function call."
+            )
+
+        def _convert_items_to_upsert_format(_items):
+            _upsert_items: List[DataRowUpsertItem] = []
+            for item in _items:
+                # enforce current dataset's id for all specs
+                item['dataset_id'] = self.uid
+                if 'key' not in item:
+                    key = {'type': 'AUTO', 'value': ''}
+                elif isinstance(item['key'], UniqueId):
+                    key = {'type': 'ID', 'value': item['key'].key}
+                    del item['key']
+                elif isinstance(item['key'], GlobalKey):
+                    key = {'type': 'GKEY', 'value': item['key'].key}
+                    del item['key']
+                else:
+                    raise ValueError(
+                        f"Key must be an instance of UniqueId or GlobalKey, got: {type(item['key']).__name__}"
+                    )
+                item = {
+                    k: v for k, v in item.items() if v is not None
+                }  # remove None values
+                _upsert_items.append(DataRowUpsertItem(payload=item, id=key))
+            return _upsert_items
+
+        specs = _convert_items_to_upsert_format(items)
+        chunks = [
+            specs[i:i + self.__upsert_chunk_size]
+            for i in range(0, len(specs), self.__upsert_chunk_size)
+        ]
+
+        def _upload_chunk(_chunk):
+            return self._create_descriptor_file(_chunk, is_upsert=True)
+
+        file_upload_thread_count = 20
```

Comment:
Personally, I would expose this value in the method signature and provide a reasonable default.